src/_helpers/auth.ts

```diff
@@ -0,0 +1,92 @@
+import jwt from 'jsonwebtoken';
+import { logger } from "../index";
+import { currentConfig } from './config';
+import { clone } from "./clone";
+
+import { AuthenticateSuccessResponse } from '../_models/AuthenticateSuccessResponse';
+import { User } from '../_models/User';
+
+export function authenticate(username: string, password: string): Promise<AuthenticateSuccessResponse> {
+    return new Promise<AuthenticateSuccessResponse>((resolve, reject) => {
+        console.log(currentConfig.users);
+        let userFound = false;
+        currentConfig.users.forEach((user_original) => {
+            let user = clone(user_original);
+            if(username === user.username) {
+                userFound = true;
+                if(password !== user.password) {
+                    reject(new Error('Password is incorrect'));
+                }
+                delete user["password"];
+                jwt.sign({user}, currentConfig.security.jwt_private_key, { expiresIn: '2 days' }, (err, token) => {
+                    if(err) { reject(err) }
+                    resolve({
+                        access_token: token,
+                        user: user
+                    });
+                });
+                return true;
+            }
+        });
+        if(!userFound) reject(new Error('User not found'));
+    });
+}
+
+export function validateAccessToken(access_token: string): Promise<User> {
+    return new Promise<User>((resolve, reject) => {
+        jwt.verify(access_token, currentConfig.security.jwt_private_key, (err, decoded) => {
+            console.log(decoded);
+            if(err) {
+                reject(err);
+            }
+            resolve(decoded.user);
+        });
+    });
+}
+
+export function getUsersList(removePassword = false): User[] {
+    let users = clone(currentConfig.users);
+    if(removePassword) {
+        users.forEach((user) => {
+            delete user["password"];
+        });
+    }
+    return users;
+}
+
+export function addUser(user: User) {
```

Comment:
I really like the new authentication workflow with the access token. If we make TA now quite secure, we should also move away from storing passwords as plaintext ;-)
https://www.loginradius.com/blog/async/hashing-user-passwords-using-bcryptjs/